web/server/codechecker_server/api/report_server.py

```diff
@@ -1722,7 +1722,7 @@ def getLinesInSourceFileContents(self, lines_in_files_requested, encoding):
                 sourcefile = session.query(File).get(lines_in_file.fileId)
                 cont = session.query(FileContent).get(sourcefile.content_hash)
                 lines = zlib.decompress(
-                    cont.content).decode('utf-8').split('\n')
+                    cont.content).decode('utf-8', 'replace').split('\n')
```

Comment:
We use `decode` function in multiple places in this file. Don't we need to pass this extra second parameter (`'replace'`) everywhere?

Reply:
To be honest, I'm a little afraid of changing it at all occurrences. Even in this case I don't know if `'ignore'` should be better, because we use that at the other parts of the program. And sometimes it might be meaningful how to handle special characters, for example when we compute a bug hash from the file content at these lines.

Reply:
Yes it can cause hash mismatch if the hash is generated from this string after storage where `replace` is used compared to where it is usually done after analysis, I think ther the `get_line` function is used 
https://github.com/Ericsson/codechecker/blob/4e55e4c0398ca213bb9f3e3d40052addaeb60107/codechecker_common/util.py#L36 where characters with errors are ignored by default.

We talked about to use `replace` so at least the user can see if there are some non utf-8 valid characters.
If we want to follow that we can use `ignore` here too, but I do not think this should be used for hash generation.